clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -284,6 +425,65 @@ private void refreshAllValueSchema() throws VeniceClientException {
     }
   }
 
+  private void refreshAllUpdateSchemas() throws VeniceClientException {
+    valueSchemaMap.keySet()
+        .forEach(
+            valueSchemaId -> valueSchemaIdToUpdateSchemaMap
+                .computeIfAbsent(valueSchemaId, this::fetchUpdateSchemaEntry));
+  }
+
+  private DerivedSchemaEntry fetchUpdateSchemaEntry(int valueSchemaId) throws VeniceClientException {
+    String requestPath = TYPE_GET_UPDATE_SCHEMA + "/" + storeClient.getStoreName() + "/" + valueSchemaId;
+    SchemaResponse schemaResponse = fetchSingleSchemaResponse(requestPath);
+
+    if (schemaResponse == null) {
+      return null;
+    }
+
+    DerivedSchemaEntry updateSchemaEntry =
+        new DerivedSchemaEntry(valueSchemaId, schemaResponse.getDerivedSchemaId(), schemaResponse.getSchemaStr());
+    if (!updateSchemaEntry.getSchema().getType().equals(Schema.Type.RECORD)) {
+      throw new InvalidVeniceSchemaException("Update schema can only be record schema.");
+    }
+
+    LOGGER.info(
+        "[Store: {}] Got update schema id: {}-{}; schema: {}",
+        storeName,
+        valueSchemaId,
+        updateSchemaEntry.getId(),
+        updateSchemaEntry.getSchema());
+
+    return updateSchemaEntry;
+  }
+
+  // It would be better to use StoreStateReader and return a Store object, but it needs a lot of refactoring due to the
+  // module structure and dependencies of StoreJSONSerializer.
+  private boolean storeHasPartialUpdateEnabled() {
+    JsonNode storeState = fetchStoreState();
+    if (storeState == null) {
+      return false;
+    }
+    return storeState.get("writeComputationEnabled").asBoolean(false);
```

Comment:
This piece of logic is tricky..
Shall we just enhance the logic in `get_update_schema` endpoint to return `null` when `write-compute` is disabled instead of making two calls from client?
With ^, we can get rid of this `store_state` call and this tricky check here.
What do you think?

Reply:
Right now, we don't have an endpoint to fetch all update schemas. So, the calls are one-by-one for a value schema id and we would be relying on `n` return values instead. I could add an endpoint to fetch all update schemas too

Reply:
Discussed offline and added the endpoint to return all update schemas